WalletWasabi.Fluent/ViewModels/SearchBar/Sources/TransactionsSearchSource.cs

```diff
@@ -117,6 +119,28 @@ private static IEnumerable<ISearchItem> Search(string query)
 	private static IEnumerable<(WalletViewModel, HistoryItemViewModelBase)> Filter(string queryStr)
 	{
 		return Flatten(GetTransactionsByWallet())
-			.Where(tuple => ContainsId(tuple.Item2, queryStr));
+		.Where(tuple => IsBitcoinAddress(tuple.Item1.WalletModel.Network, queryStr) ?
+			ContainsDestinationAddress(tuple, queryStr) :
+			ContainsId(tuple.Item2, queryStr));
+	}
+
+	private static bool ContainsDestinationAddress((WalletViewModel, HistoryItemViewModelBase) tupleWalletHistoryItem, string queryStr)
+	{
+		var txid = tupleWalletHistoryItem.Item2.Transaction.Id;
+		return tupleWalletHistoryItem.Item1.WalletModel.Transactions.GetDestinationAddresses(txid)
+			   .Contains(BitcoinAddress.Create(queryStr, tupleWalletHistoryItem.Item1.WalletModel.Network));
+	}
+
+	private static bool IsBitcoinAddress(Network network, string queryStr)
+	{
+		try
+		{
+			BitcoinAddress.Create(queryStr, network);
+			return true;
+		}
+		catch (FormatException)
+		{
+			return false;
+		}
```

Comment:
The following suggestion improves readability and simplifies the code so the bitcoin address doesn't have to be recreated.
Don't forget to add `using System.Diagnostics.CodeAnalysis;`

```suggestion
		.Where(tuple => TryGetBitcoinAddress(tuple.Item1.WalletModel.Network, queryStr, out var address) ?
			ContainsDestinationAddress(tuple.Item1, tuple.Item2, address) :
			ContainsId(tuple.Item2, queryStr));
	}

	private static bool ContainsDestinationAddress(WalletViewModel walletViewModel, HistoryItemViewModelBase historyItem, BitcoinAddress address)
	{
		var txid = historyItem.Transaction.Id;
		return walletViewModel.WalletModel.Transactions.GetDestinationAddresses(txid).Contains(address);
	}

	private static bool TryGetBitcoinAddress(Network network, string queryStr, [NotNullWhen(true)] out BitcoinAddress? address)
	{
		address = null;
		try
		{
			address = BitcoinAddress.Create(queryStr, network);
			return true;
		}
		catch (FormatException)
		{
			return false;
		}
```